data/state/factory/peerAccountCreator.go

```diff
@@ -12,7 +12,7 @@ func NewPeerAccountCreator() state.AccountFactory {
 }
 
 // CreateAccount calls the new Account creator and returns the result
-func (c *PeerAccountCreator) CreateAccount(address state.AddressContainer, tracker state.AccountTracker) (state.AccountHandler, error) {
+func (pac *PeerAccountCreator) CreateAccount(address state.AddressContainer, tracker state.AccountTracker) (state.AccountHandler, error) {
```

Comment:
i think you can refactor this and just **return state.NewPeerAccount(address, tracker)**